msmbuilder/featurizer/multichain.py

```diff
@@ -0,0 +1,510 @@
+# Author: Brooke Husic <brookehusic@gmail.com>
+# Contributors: 
+# Copyright (c) 2016, Stanford University and the Authors
+# All rights reserved.
+
+from __future__ import print_function, division, absolute_import
+
+import warnings
+
+import mdtraj as md
+from mdtraj.utils import ensure_type
+from mdtraj.utils.six import string_types
+from mdtraj.utils.six.moves import xrange
+import numpy as np
+import itertools
+
+from msmbuilder import libdistance
+from . import Featurizer, zippy_maker, dict_maker, featurize_all
+
+
+class LigandFeaturizer(Featurizer):
+    """Base class for objects that featurize protein-ligand trajectories.
+
+    Notes
+    -----
+    At the bare minimum, a featurizer must implement the `partial_transform(traj)`
+    member function.  A `transform(traj_list)` for featurizing multiple
+    trajectories in batch will be provided.
+    """
+
+
+    def __init__(self, protein_chain='auto', ligand_chain='auto',
+                 reference_frame=None):
+        self.protein_chain = protein_chain
+        self.ligand_chain = ligand_chain
+        self.reference_frame = reference_frame
+
+        if reference_frame is None:
+            raise ValueError("Please specify a reference frame")
+
+        if reference_frame.n_frames is not 1:
+            raise ValueError("Reference frame must be a single frame")
+
+        if reference_frame.top.n_chains < 2:
+            raise ValueError("Topology must have at least two chains")
+
+        self.contacts = self._guess_chains(self.reference_frame)
+        self.p_residue_offset = self._get_residue_offset(self.reference_frame,
+                                                         self.protein_chain)
+        self.l_residue_offset = self._get_residue_offset(self.reference_frame,
+                                                         self.ligand_chain)
+        self.p_atom_offset = self._get_atom_offset(self.reference_frame,
+                                                   self.protein_chain)
+        self.l_atom_offset = self._get_atom_offset(self.reference_frame,
+                                                   self.ligand_chain)
+
+
+    def _guess_chains(self, traj):
+        if self.protein_chain == 'auto' or self.ligand_chain == 'auto':
+            chain_dict = {}
+        for i in range(traj.top.n_chains):
+            # each entry in the chain_dict is a list:
+            # [number of atoms, has a CA, has <100 atoms]
```

Comment:
Again, of the ligands I'm working with is glutamine, which does have an amino acid name. Like I said below, seems better to allow for the rare case than to get rid of the option.